localstack/services/apigateway/apigateway_listener.py

```diff
@@ -507,11 +507,13 @@ def get_lambda_event_request_context(method, path, data, headers,
     source_ip = headers.get('X-Forwarded-For', ',').split(',')[-2].strip()
     integration_uri = integration_uri or ''
     account_id = integration_uri.split(':lambda:path')[-1].split(':function:')[0].split(':')[-1]
+    domain_name = get_domain_from_path(path)
```

Comment:
oh..hmm...
this makes sense.
I will give it a try

Reply:
@whummer 
Actually, this might not work because the url we have with localstack is
"http://localhost:4566/restapis/<rest_api_id>/localstack/_user_request_/path"

Reply:
Yes, good point. I think we have 2 options here. We could either:
1) Return the actual/real domain name that was used for the invocation (which would be `localhost` in most cases), or
2) Return the virtual domain name that the API is associated with (e.g., `api123.execute-api.localhost.localstack.cloud`, which is more in line with the behavior in AWS.)

I'm more leaning towards option (2) above, which is represented by the code I've shared above.